### ISSUE TITLE
Documentation update and translation with transifex

### ISSUE THREAD

**Issue by privatereese** [nextcloud/documentation#216]
Would appreciate an update of the documentation and translation via transifex.

Major goal is an update of the documentation.
Minor goal is transifex translation.

Would start with an update of the english documentation and then at least create a german translation via transifex.
As I am new to nextcloud i would appreciate every help i can get.
I assume that this should be handled with high priority as it helps spreading nextcloud to users and could ease handle and  supporting effort.

**Comment by esprit-libre** [comment]
Hello,
also interested for creating and contributing to french translation of user manual (admin and dev manual doesn't need to be translated in french).
My need is to share this doc to my customers, mostly french speaking users.
Any chance to have this?
Best regards.

**Comment by dhebert91** [comment]
Hello,
I'm also interested with a french translation of the user manual and i can contribute to it.
Regards

**Comment by guidtz** [comment]
@esprit-libre @dhebert91 cela vous intéresserait de mettre en place cette traduction et une fois qu'elle est en place faire un pull request pour la proposer ? On pourrait partir sur la version 12 et se répartir le travail de traduction / relecture. 
Je suis dans le même besoins de fournir une doc utilisateur en français à mes clients et pas obligatoirement le temps de tout réécrire.
J'ai commencé à regarder un peu, il faut modifier les fichiers rst et générer le pdf, voir adapter les screenshots si besoins. 
guidtz 
@esprit-libre on c'est déjà croisé lors des RRLL à Nantes il me semble : Guillaume Chéramy, société Hisyl faisant partie d'Alliance Libre

**Comment by pierreozoux** [comment]
@jancborchardt how can we get started on that?

I'm seeing [that](https://howto.disroot.org/en/nextcloud/introduction) also, and I'm a bit sad that this doc is not on Nextcloud to collaborate upstream :)

I think we could do a lot better job here, and would be nice to start to work on that!

I have people asking on how to help, and I think here is a good place!

**Comment by jancborchardt** [comment]
@nextcloud/documentation and @muppeth from Disroot: Get together please :)

**Comment by esprit-libre** [comment]
and also [this](https://fr.wikibooks.org/wiki/Manuel_utilisateur_Nextcloud)

**Comment by frju365** [comment]
any update ?

**Comment by MorrisJobke** [comment]
cc @jospoortvliet

**Comment by muppeth** [comment]
@frju365 from my side i can say together with some community members responsible for most howtos and translations we are planning a (first) meeting where we discuss how to coordinate efforts in disroot howtos and tutorials and one of the subjects is how we can contribute upstream.

**Comment by dacoex** [comment]
Maybe this could serve as some help:
https://docs.readthedocs.io/en/stable/guides/manage-translations.html

It would be great to bring these translations more into synch.
Just look into these folders:

- https://github.com/nextcloud/documentation/tree/master/user_manual_pt-br/pim

- https://github.com/nextcloud/documentation/tree/master/user_manual_de/pim
- https://github.com/nextcloud/documentation/tree/master/user_manual/pim

**Comment by pierreozoux** [comment]
I need to work on this too!

@frju365 did you finish the french translation?

I have a plan that I'll start to execute the coming days:

Alors, c'est un peu complexe cette histoire!

The doc is already on the Nextcloud instances
 /settings/help?mode=user
(this is an iframe --> /core/doc/user/index.html )

We need to:
 - [ ] [configure transifex](https://docs.transifex.com/integrations/sphinx-doc) for this repo
 - [ ] check https://github.com/paquerette-orga/documentation and merge it here.
 - [ ] add'i18n, probably [here](https://github.com/nextcloud/server/blob/master/apps/settings/lib/Controller/HelpController.php#L77)
 - [ ] understand how this doc is shipped into Nextcloud server code. @MorrisJobke or @jancborchardt could you help here? 

Thanks a lot!

**Comment by jancborchardt** [comment]
cc'ing @rullzer for translations (or he will know the right person) instead of Morris or me. :)

**Comment by dhebert91** [comment]
Thank's @pierreozoux to revive this project.
https://github.com/paquerette-orga/documentation is in stand by because the original user manual lack of completion and is not updated ( [What’s new for users in Nextcloud 13](https://docs.nextcloud.com/server/13/user_manual/whats_new.html) has exactly the same content as [What’s new for users in Nextcloud 17](https://docs.nextcloud.com/server/17/user_manual/whats_new.html) )
I am interested to participate; is it possible to have a phone call (i have no rendez-vous tomorrow).
My phone number: +33 6 07 86 09 42

**Comment by pierreozoux** [comment]
@dhebert91 I will!

people, you can already translate here:
https://www.transifex.com/indiehosters/nextcloud-user-documentation/languages/

I'll PR some code to the CI. I still have a question, where is the .de translation online? Not really clear at the moment.

@rullzer we have momentum again regarding this issue, keep it while it is warm ;)

**Comment by pierreozoux** [comment]
I configured transifex, if you want to be part of the org, please send me your [email](pierre@ozoux.net), or request it there, but the interface is not that great, so I might not see it.

@nextcloud/documentation I think it should be under Nextcloud organizartion. Whenever you are ready, I can give it to you (no idea how to do in transifex..).

**Comment by pierreozoux** [comment]
@johkoenig could you move the german translation to transifex?
@richardnixonafj could you move pt-br translation to transifex too?

I'll now work on setting up the automation to autmatically compile the docuemntation including translation from transifex.

**Comment by p-bo** [comment]
@pierreozoux I completed translation into Czech (sent you an e-mail about this too). What next?

**Comment by pierreozoux** [comment]
Yes saw your email :/ sorry for delay, a bit busy currently ;)

What is your need? Do you want it shipped with Nc itself or on the doc website?

**Comment by p-bo** [comment]
@pierreozoux sorry for my impatience ;)

Must admit, that haven't idea how much work is needed for publish it on doc and/or ship with NC itself, but at the end, ideally both.
But for start, it would be great to have it rendered on doc website to see, how translation fit (and as I will tweak it, to be able see result of tweaks and, to iterate quicker, would it be possible to refresh regularly please?).
Regarding shipping with Nc itself, don't know, if there is now possible to have documentation in more languages - when not, there is no need for waiting to final tweaked translation and laying groundwork to it could be done in parallel, I hope (and plea for it).
I know, I ask for lot of work from you (and others?), but as mere translator, non-developer, I unfortunately cannot help more than to tirelessly translate ;)

Thanks for you help

**Comment by pierreozoux** [comment]
No worries :) free software dynamics are complexe ;) we (technical people) have to help others to be independent and be able to contribute without us being on the way.

**Comment by pierreozoux** [comment]
Sorry for the delay :/

https://doc-nextcloud-fr.indie.host/com/cs/
https://doc-nextcloud-fr.indie.host/release/cs/
https://doc-nextcloud-fr.indie.host/com/fr/
https://doc-nextcloud-fr.indie.host/release/fr/

I have some automation I'll PR here. In the mean time, feel free to ping me here or by email.

**Comment by p-bo** [comment]
Great! Thanks for that.
So started reviewing how translated strings fits in place of original ones. During this I revealed, that source strings are probably a bit outdated - for example in "Synchronizing with iOS" chapter of Groupware section, there is only 8 steps in original text (latest version) and in both translations (fr, cs), there are 10 of them.
But it will be be solved by automation, you mentioned - right?

What are our plans to have these translations merged into upstream?
From previous communication with developers, they didn't reacted to your offer to merge translation on Transifex (which is unfortunate, because translators of Nextcloud apps itself are very active, they only don't know about docs translation probably - merging would help to make it more visible).
I can kindly lobbying toward developers only from position of someone, who translated/fixed most of strings in whole Czech translation of strings Nextcloud apps, if it helps.

@johkoenig
@richardnixonafj
Please lets join our efforts on Transifex docs translation projects - thanks! :)

**Comment by johkoenig** [comment]
Hey guys,
I don't know how my name ended up in this ticket, I didn't work on this matter ever before.
Please take my apologies that my free time to work on open source projects is currently very limited for personal reasons. So, I will not be able to support you on this. 
Thanks for your understanding

**Comment by pierreozoux** [comment]
@p-bo if it is a bit outdated, it has to be fixed in english first.
The automation is just about how the stings are sent to transifex, and how the resources are then put together and built. See #1824 but this is really technical, the plumbing if you will.

I sent the request to move the transifex project to nextcloud.

I think @skjnldsv is the best person to help here. I propose that once the PR is merged, we can close this issue. If the documentation has to be updated, then open specific issues.

**Comment by pierreozoux** [comment]
@jancborchardt @jospoortvliet 
Can we get an update on this?
We spent days working on it, and we have no feedback from you.
And now I see that the name of our work was changed to be prefixed by "stop" ??

https://www.transifex.com/nextcloud/nextcloud-user-documentation/

I'm sure there is a good reason, but please do communicate, I have no idea what is happening, and as I spent days (not the only one) I think we deserve some explanation or some roadmap of what is going on.

Thanks.

**Comment by p-bo** [comment]
@jancborchardt @jospoortvliet I second to that - please for sake of better user experience for non-English speakers, don't throw translation of _user_ documentation away please…

**Comment by p-bo** [comment]
@pierreozoux it seems, that current status is here: https://github.com/nextcloud/documentation/issues/1832#issuecomment-613303930

**Comment by jospoortvliet** [comment]
@p-bo that is indeed the current state and roadmap: we currently have no plans as we have no time for it. The best approach would be a way to connect Transifex to our current docs as that would not disrupt our workflow. We don't know if it is possible and if it is, how to do it. We'd be extremely grateful to anyone willing to do the research and perhaps point out how to do it.

**Comment by skjnldsv** [comment]
> Can we get an update on this?
> We spent days working on it, and we have no feedback from you.

19 is about to be released. We're all completely swamped under all sort of things! :(
Sorry about this! Thanks for the awesome work here, we'll dive into this as soon as we can!

**Comment by pierreozoux** [comment]
@skjnldsv ok, sorry, maybe my rant was a bit out of context.

We did coordinate work to:
 - put all the doc into transifex
 - translate it in french and czech
 - automate the build of the doc - the [PR is here](https://github.com/nextcloud/documentation/pull/18240)
 - [make the use of the translated documentation in Nc with the l10n of the user](https://github.com/nextcloud/server/pull/17870)

We spent days doing this. We created a transfix repo for that.

Then, we transferred the repo to Nc, with the hope that it will land soonish (the automatic build PR has now 3 months).

Currently we are frustrated because:
 - the name of the [transifex repo](https://www.transifex.com/nextcloud/nextcloud-user-documentation/) has a "stop" in it
 - people that apply to translate are refused
 - I can't myself build the documentation any more, as I don't have the transifex token, and I'm not sure I can "fork" a transifex repo (the one I created in the first place). (this one is very frustrating)

I understand it is open source and all, but here I'm working, I'm not just asking you to do things for me for free, we need you to merge, that's true.

I don't need it now, I'm patient. I also understand that you are understaffed, or overwhelmed, especially now, it is a complicated time.

So please, finish 19th release, take your time, no burn out :)

Then, let's come back to this issue, I think we have a momentum, let's not loose it :) I'm available for you here, over [email](pierre@ozoux.net), [chat](https://chat.indie.host/) or phone.

Actually, I could merge the PR myself, if I had the transifex token. So if you want this to get faster:
 - make me admin of transifex
 - give me rights for this build on the CI

Then I can do everything myself, I will not need to bother you :)

There is a chance that my PR breaks the Nextcloud build, as I don't know how it works, but let's wait 19 to be released, and then I could safely break it :)

Take care, no burn out, and give me rights :) Cheers!

**Comment by richardnixonafj** [comment]
> Great! Thanks for that.
> So started reviewing how translated strings fits in place of original ones. During this I revealed, that source strings are probably a bit outdated - for example in "Synchronizing with iOS" chapter of Groupware section, there is only 8 steps in original text (latest version) and in both translations (fr, cs), there are 10 of them.
> But it will be be solved by automation, you mentioned - right?
> 
> What are our plans to have these translations merged into upstream?
> From previous communication with developers, they didn't reacted to your offer to merge translation on Transifex (which is unfortunate, because translators of Nextcloud apps itself are very active, they only don't know about docs translation probably - merging would help to make it more visible).
> I can kindly lobbying toward developers only from position of someone, who translated/fixed most of strings in whole Czech translation of strings Nextcloud apps, if it helps.
> 
> @johkoenig
> @richardnixonafj
> Please lets join our efforts on Transifex docs translation projects - thanks! :)

sorry for delay, still need help with portuguese - BR  translations?

**Comment by p-bo** [comment]
@richardnixonafj right now, there is need to sorting out - see posts above. Then help will be very welcomed! :)

**Comment by dhebert91** [comment]
I did work on French translation with @pierreozoux and i invoke @jospoortvliet to help us to find the good person in the nextcloud team able to unblock the situation.

**Comment by pierreozoux** [comment]
@skjnldsv now that 19 has been released, can we get this moving?

**Comment by jancborchardt** [comment]
cc @rullzer @MorrisJobke @jospoortvliet for coordination and unblocking on this one. :)

**Comment by rullzer** [comment]
So this was a one time import right?

Basically we'd need to figure out some things

1. Sync from github => transifex
2. Sync translations from transifex => github
3. Find a way to actually build translations

I must admit I have no idea about all 3 points.
If somebody know how to do this? Then we can move this on.

**Comment by pierreozoux** [comment]
@rullzer I already addressed the 3 points here:
https://github.com/nextcloud/documentation/pull/1824

**Comment by pierreozoux** [comment]
@rullzer @jancborchardt @MorrisJobke @jospoortvliet 

It has been more than 5 months I did create this [PR](https://github.com/nextcloud/documentation/pull/1824) to automatize the build of documentation and their translations with transifex.
I did create the transifex repo, and with the help of some people we did put the french and czec translation.

I need this translation and I'm blocked by you, I can't build the translated documentation anymore. I'm very frustrated, and I guess you understand why.

Can you please take a look at this PR, and let me know that you'll work on it? There is just one more step to accomplish, add the transfiex API token to the build to make it automatic. (if I'd be admin of the Nextcloud transifex org, I could do it myself if you prefer).
If you can't do that, I kindly ask you to transfer me back [this transifex](https://www.transifex.com/nextcloud/nextcloud-user-documentation/) project.

thanks.

**Comment by skjnldsv** [comment]
@pierreozoux done!
Sorry again! Are you on our community chat? Would be nice so we can better communicate! :)

**Comment by rakekniven** [comment]
@skjnldsv Sorry, but I am busy at the moment.
Shortd words: I do not know which direction to go for the documentation.

In 7 days my holidays should start *hopefully* and I look into it more deeply.

**Comment by skjnldsv** [comment]
@rakekniven no problem :)
I've taken over and we're about to finish implementing this! Thanks for the feedback and enjoy vacations!! :rocket: :hugs:
